certora/specs/SocialRecoveryModule.spec

```diff
@@ -1,26 +1,176 @@
+    function _.isModuleEnabled(address module) external => safeIsModuleEnabled(module) expect bool ALL;
+    function _.isOwner(address owner) external => safeIsOwner(calledContract, owner) expect bool ALL;
 }
 
-// A setup function that requires Safe contract to enabled the Social Recovery
-// Module.
+// Ghost variable to store the guardian mapping of a Safe.
+ghost mapping(address => mapping(address => address)) ghostGuardians {
+    init_state axiom
+        forall address safeAccount.
+        forall address guardian.
+            ghostGuardians[safeAccount][guardian] == 0;
+}
+hook Sload address value guardianStorageContract.entries[KEY address safeAccount].guardians[KEY address guardian] {
+    require ghostGuardians[safeAccount][guardian] == value;
+}
+hook Sstore guardianStorageContract.entries[KEY address safeAccount].guardians[KEY address guardian] address value {
+    ghostGuardians[safeAccount][guardian] = value;
+}
+
+// Ghost variable to store the guardian count of a Safe.
+ghost mapping(address => mathint) ghostGuardianCount {
+    init_state axiom forall address safeAccount. ghostGuardianCount[safeAccount] == 0;
+}
+hook Sload uint256 value guardianStorageContract.entries[KEY address safeAccount].count {
+    require ghostGuardianCount[safeAccount] == to_mathint(value);
+}
+hook Sstore guardianStorageContract.entries[KEY address safeAccount].count uint256 value {
+    ghostGuardianCount[safeAccount] = value;
+}
+
+// Ghost variable to store the threshold of a Safe.
+ghost mapping(address => mathint) ghostThreshold {
+    init_state axiom forall address safeAccount. ghostThreshold[safeAccount] == 0;
+}
+hook Sload uint256 value guardianStorageContract.entries[KEY address safeAccount].threshold {
+    require ghostThreshold[safeAccount] == to_mathint(value);
+}
+hook Sstore guardianStorageContract.entries[KEY address safeAccount].threshold uint256 value {
+    ghostThreshold[safeAccount] = value;
+}
+
+// A summary function to check if a module is enabled in the Safe contract.
+function safeIsModuleEnabled(address module) returns bool {
+    return safeContract.isModuleEnabled(module);
+}
+
+// A summary function to check if an address is an owner in the Safe contract.
+function safeIsOwner(address calledContractAddress, address owner) returns bool {
+    assert calledContractAddress == safeContract; // This checks that the contract is Safe Smart Account.
+    return safeContract.isOwner(owner);
+}
+
+// A setup function that requires Safe contract to enabled the Social Recovery Module.
 function requireSocialRecoveryModuleEnabled() {
     require(safeContract.isModuleEnabled(currentContract));
 }
 
-// This is a dummy rule to verify the Safe Contract Setup with the Social Recovery
-// Module is working as intended.
-rule recoveryModuleCanBeDisabled {
-    env e;
-    address prevModule;
+// A helper function to verify that two addresses are not the same.
+function notSameAddress(address a, address b) {
```

Comment:
We should reflect that this `require`s something inside. The function name suggests it returns a boolean depending on whether or not they are equal.

Also, I think this function is not useful. It just hides what it is doing without providing any useful abstraction.

Reply:
Removed 👍🏾